galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java

```diff
@@ -95,54 +95,61 @@ public void processDataArray(JsonArray jsonArray, String action) throws Internal
                         ServletError error = new ServletError(GAL5026_UNSUPPORTED_RESOURCE_TYPE,kind);
                         throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
                 }
-            }catch(InternalServletException s){
+            } catch(InternalServletException s){
                 errors.add(s.getMessage());
-            }catch(Exception e){
+            } catch(Exception e){
                 ServletError error = new ServletError(GAL5000_GENERIC_API_ERROR);
                 throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
             }
         }
     }
     
-    public void processGalasaProperty (JsonObject resource, String action) throws InternalServletException{
+    public void processGalasaProperty(JsonObject resource, String action) throws InternalServletException{
         try{
             String apiversion = resource.get("apiVersion").getAsString();
             String expectedApiVersion = GalasaProperty.DEFAULTAPIVERSION;
-            if (apiversion.equals(expectedApiVersion)){
+            if (apiversion.equals(expectedApiVersion)) {
                 GalasaProperty galasaProperty = gson.fromJson(resource, GalasaProperty.class);           
-                if (galasaProperty.isPropertyValid()){
+                if (galasaProperty.isPropertyValid()) {
                     CPSFacade cps = new CPSFacade(framework);
                     CPSNamespace namespace = cps.getNamespace(galasaProperty.getNamespace());
                     CPSProperty property = namespace.getPropertyFromStore(galasaProperty.getName());
```

Comment:
Won't this be null now, if the property isn't in the store ?

Reply:
That is exactly what it is doing the only time it does not do that is if the namespace is hidden to force an exception